app/src/main/java/com/mobileproto/hireddit/hireddit/GetWordsAsync.java

```diff
@@ -0,0 +1,120 @@
+package com.mobileproto.hireddit.hireddit;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.google.common.io.CharStreams;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import io.indico.Indico;
+import io.indico.network.IndicoCallback;
+import io.indico.results.IndicoResult;
+import io.indico.utils.IndicoException;
+
+;
+
+/**
+ * Created by lwilcox on 11/5/2015.
+ */
+public class GetWordsAsync extends AsyncTask<Void, Void, ArrayList<String>>
+{
+    private String spokenString;
+    private String importantWords;
+    private Context context;
+    public ArrayList<String> wordList = new ArrayList<>();
+    public ArrayList<String> allComments;
+    public String postComment;
+    public TextView commentText;
+    public Indico indico;
+
+
+    public GetWordsAsync(String spokenString, String importantWords, Context context, TextView commentText)
```

Comment:
instead of passing in a `TextView`, is it not better to have the `callback` callback to the text view instead of this class, where `commentText` is given to the fragment? i think all view related stuff is supposed to primarily stay in fragments, correct me if i'm wrong